src/quick-lint-js/utf-16.h

```diff
@@ -0,0 +1,43 @@
+// quick-lint-js finds bugs in JavaScript programs.
+// Copyright (C) 2020  Matthew Glazar
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+#ifndef QUICK_LINT_JS_UTF_16_H
+#define QUICK_LINT_JS_UTF_16_H
+
+#include <optional>
+#include <string>
+
+namespace quick_lint_js {
+#if defined(_WIN32)
+class mbargv {
+ public:
+  explicit mbargv(int argc, wchar_t **wargv);
+  ~mbargv();
```

Comment:
Follow the Rule of Five (or the Rule of Three): if you declare a destructor, you should also declare a copy constructor, a move constructor, a copy assignment operator, and a move assignment operator.

To simplify things, you could delete the copy functions to satisfy the rule:
```cpp
mbargv(const mbargv &) = delete;
mbargv &operator=(const mbargv &) = delete;
```